java/ingest/ingest-batcher-submitter/src/test/java/sleeper/ingest/batcher/submitter/IngestBatcherSubmitterLambdaIT.java

```diff
@@ -0,0 +1,329 @@
+                            fileRequest(TEST_BUCKET + "/test-directory/nested-1/test-file-1.parquet"));
+        }
+    }
+
+    @Nested
+    @DisplayName("Store files from multiple file paths")
+    class StoreFilesFromMultipleFilePaths {
+        @Test
+        void shouldStoreFilesWhenJsonHasMultipleFilePaths() {
+            // Given
+            uploadFileToS3("test-file-1.parquet");
+            uploadFileToS3("test-file-2.parquet");
+            String json = "{" +
+                    "\"files\":[" +
+                    "\"test-bucket/test-file-1.parquet\"," +
+                    "\"test-bucket/test-file-2.parquet\"" +
+                    "]," +
+                    "\"tableName\":\"test-table\"" +
+                    "}";
+
+            // When
+            lambda.handleMessage(json, RECEIVED_TIME);
+
+            // Then
+            assertThat(store.getAllFilesNewestFirst())
+                    .containsExactly(
+                            fileRequest("test-bucket/test-file-2.parquet"),
+                            fileRequest("test-bucket/test-file-1.parquet"));
+        }
+
+        @Test
+        void shouldSkipFileIfNotFound() {
+            // Given
+            uploadFileToS3("test-file-2.parquet");
+            String json = "{" +
+                    "\"files\":[" +
+                    "\"test-bucket/not-found.parquet\"," +
+                    "\"test-bucket/test-file-2.parquet\"" +
+                    "]," +
+                    "\"tableName\":\"test-table\"" +
+                    "}";
+
+            // When
+            lambda.handleMessage(json, RECEIVED_TIME);
+
+            // Then
+            assertThat(store.getAllFilesNewestFirst())
+                    .containsExactly(
+                            fileRequest("test-bucket/test-file-2.parquet"));
+        }
+    }
+
+    @Nested
+    @DisplayName("Handle errors in messages")
+    class HandleErrorsInMessage {
+
+        @Test
+        void shouldIgnoreAndLogMessageWithInvalidJson() {
+            // Given
+            String json = "{";
+
+            // When
+            lambda.handleMessage(json, RECEIVED_TIME);
+
+            // Then
+            assertThat(store.getAllFilesNewestFirst()).isEmpty();
+        }
+
+        @Test
+        void shouldIgnoreAndLogMessageIfTableDoesNotExist() {
+            // Given
+            String json = "{" +
+                    "\"files\":\"[" + TEST_BUCKET + "/test-file-1.parquet\"]," +
+                    "\"tableName\":\"not-a-table\"" +
+                    "}";
+
+            // When
+            lambda.handleMessage(json, RECEIVED_TIME);
+
+            // Then
+            assertThat(store.getAllFilesNewestFirst()).isEmpty();
+        }
+
+        @Test
+        void shouldIgnoreMessageIfFileDoesNotExist() {
+            // Given
+            String json = "{" +
+                    "\"files\":\"[" + TEST_BUCKET + "/not-exists.parquet\"]," +
```

Comment:
This JSON is also invalid because of a double quote before the start of the array.